server/src/main/java/io/crate/expression/scalar/ExtractFunctions.java

```diff
@@ -123,6 +123,37 @@ public static void register(ScalarFunctionModule module) {
                     new UnaryScalar<Number, Period>(signature, boundSignature, DataTypes.INTERVAL, function::apply)
             );
         }
+        // extract(epoch from ...) is different as is returns a `double precision`
+        module.register(
+            Signature.scalar(
+                functionNameFrom(EPOCH),
+                DataTypes.INTERVAL.getTypeSignature(),
+                parseTypeSignature("double precision")
+            ),
+            (signature, boundSignature) ->
+                new UnaryScalar<>(signature, boundSignature, DataTypes.INTERVAL, ExtractFunctions::toMillis)
+        );
+    }
+
+    private static final long YEAR_IN_SECONDS = 365 * 24 * 60 * 60L;
+    private static final long MONTH_IN_SECONDS = 30 * 24 * 60 * 60L;
```

Comment:
Is this matching PostgreSQL behavior?

Reply:
Yep, that's why I added this +6h per year.

Reply:
```
{15:26}~/crate/crate:mt/epoch ✗ ➭ psql -U postgres                                                                                                                    :mt/epoch|✚1⚑1
psql (14.6)
Type "help" for help.

postgres=# SELECT extract(epoch from INTERVAL '7 years 11 months 18 days 11 hours');
     extract      
------------------
 251010000.000000
(1 row)
```